proxy/src/telemetry/metrics/histogram.rs

```diff
@@ -86,18 +85,31 @@ impl<V: Into<u64>> Histogram<V> {
 }
 
 impl<'a, V: Into<u64>> IntoIterator for &'a Histogram<V> {
-    type Item = (Bucket, u64);
-    type IntoIter = iter::Map<
-        slice::Iter<'a, (Bucket, Counter)>,
-        fn(&'a (Bucket, Counter)) -> (Bucket, u64)
+    type Item = (&'a Bucket, &'a Counter);
+    type IntoIter = iter::Zip<
+        slice::Iter<'a, Bucket>,
+        slice::Iter<'a, Counter>,
     >;
 
     fn into_iter(self) -> Self::IntoIter {
-        self.buckets.iter().map(|&(b, c)| (b.clone(), c.into()))
+        self.bounds.0.iter().zip(self.buckets.iter())
     }
 }
 
-// ===== impl Histogram =====
+// ===== impl Bounds =====
+
+impl Bounds {
+    pub fn new(buckets: &'static [Bucket]) -> Self {
```

Comment:
Does this ever get called (since you initialize a static `Bounds` in `latency.rs`)?